pinot-spi/src/main/java/org/apache/pinot/spi/stream/StreamDataDecoderImpl.java

```diff
@@ -30,6 +30,7 @@ public class StreamDataDecoderImpl implements StreamDataDecoder {
   public static final String KEY = "__key";
   public static final String HEADER_KEY_PREFIX = "__header$";
   public static final String METADATA_KEY_PREFIX = "__metadata$";
+  public static final String RECORD_SERIALIZED_VALUE_SIZE_KEY = "recordSerializedValueSize";
```

Comment:
Suggest naming it `METADATA_KEY_PREFIX + "streamMessageSize"` to be more specific. We can have it with prefix to avoid always concating them